### PR TITLE
[STABLE-v2.2] Update rimage to version 02abc5d342a3, including critical openssl3 fix

### LINKED ISSUE #9340: [BUG] rimage + openssl3 is broken in the stable-v2.2 and cavs25 branches, always have been
**Describe the bug**

rimage + openssl3 is broken in the ~stable-v2.2 and~ cavs-drop-stable branch, always has been.

This was fixed 2 years ago by rimage commit https://github.com/thesofproject/rimage/commit/73a9d7c897d3658486dc5cb5d69c7a587dfc042a 

However that critical rimage fix never made it to https://github.com/thesofproject/sof/commits/stable-v2.2/rimage. stable-v2.2 fell 6 commits short.

That's because we tested https://github.com/thesofproject/sof/commits/stable-v2.3/rimage for a long time (which does have the openssl3 fix) and then decided to go backwards to stable-v2.2. I don't remember why.

In https://github.com/thesofproject/rimage/pull/97#pullrequestreview-1009485532 @lgirdwood wrote "We need to take this for v2.2, ..." but that never happened.

The original investigation happened in 2022 in https://github.com/thesofproject/sof/issues/5917 . I didn't re-open that bug and opened a new one instead because that investigation took an incredibly long time with many red herrings so the original bug is very long and very hard to follow. One of the reasons it took so long: some people were using openssl1 (which worked) and other people were using openssl3 (which failed) but it took a long time to notice that difference.

**To Reproduce**

This affects only some SOF commits and not others and I have no idea why. Any OpenSSL expert around?

This was found in totally unrelated PR https://github.com/thesofproject/sof/pull/9336. When I fetch https://github.com/thesofproject/sof/commit/4ed65ed718109a346a7c573115c43918497729a4 which is the latest `pull/9336/merge` commit for that PR and run `git submodule update --recursive`, then the firmware never boots on my TGL Xtreme. After I cherry-pick openssl3 fix https://github.com/thesofproject/rimage/commit/73a9d7c897d3658486dc5cb5d69c7a587dfc042a, the firmware boots again every time.

(the git bisect wasn't straight-forward for unrelated reasons but it worked; you have been warned)

**Reproduction Rate**

100% for some SOF commits but 0% for other SOF commits like the current tag `v2.2.10`. No idea why some commits boot while others not.

The toolchain can also affect reproduction, see comments in unrelated #9336. I guess anything that changes the binary.

**Expected behavior**

The firmware boots.

**Impact**

Show stopper.

**Screenshots or console output**

```
juil. 31 02:39:51 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: Attempting iteration 0 of Core En/ROM load...
juil. 31 02:39:51 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: FW Poll Status: reg[0x4]=0xf010e0e successful
juil. 31 02:39:51 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: FW Poll Status: reg[0x4]=0xf010e0e successful
juil. 31 02:39:51 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: unstall/run core: core_mask = 1
juil. 31 02:39:51 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: DSP core(s) enabled? 1 : core_mask 1
juil. 31 02:39:51 tglup-mh kernel: usbcore: registered new interface driver snd-usb-audio
juil. 31 02:39:52 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: FW Poll Status: reg[0xd4]=0x0 timedout
juil. 31 02:39:52 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: FW Poll Status: reg[0x4]=0xf010f0f successful
juil. 31 02:39:52 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: FW Poll Status: reg[0x4]=0xf0f successful
juil. 31 02:39:52 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: DSP core(s) enabled? 0 : core_mask 1
juil. 31 02:39:52 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: Attempting iteration 1 of Core En/ROM load...
juil. 31 02:39:52 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: FW Poll Status: reg[0x4]=0xf010f0f successful
juil. 31 02:39:52 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: FW Poll Status: reg[0x4]=0xf010f0e successful
juil. 31 02:39:52 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: unstall/run core: core_mask = 1
juil. 31 02:39:52 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: DSP core(s) enabled? 1 : core_mask 1
juil. 31 02:39:52 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: FW Poll Status: reg[0xd4]=0x80000000 successful
juil. 31 02:39:52 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: FW Poll Status: reg[0x80000]=0x5000001 successful
juil. 31 02:39:52 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: Code loader DMA starting
juil. 31 02:39:52 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: Code loader DMA done, waiting for FW_ENTERED status
juil. 31 02:39:55 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: FW Poll Status: reg[0x80000]=0x80000012 timedout
juil. 31 02:39:55 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: hda_cl_copy_fw: timeout with rom_status_reg (0x80000) read
juil. 31 02:39:55 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: FW Poll Status: reg[0x160]=0x140000 successful
juil. 31 02:39:55 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: Code loader DMA stopped
juil. 31 02:39:55 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ------------[ DSP dump start ]------------
juil. 31 02:39:55 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: Firmware download failed
juil. 31 02:39:55 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: fw_state: SOF_FW_BOOT_IN_PROGRESS (3)
juil. 31 02:39:55 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: 0x80000012: module: ROM, state: CSE_VALIDATE_IMAGE_REQUEST, not running
juil. 31 02:39:55 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error code: 0x2c (error: signature verification failed)
juil. 31 02:39:55 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: extended rom status:  0x80000012 0x2c 0x0 0x0 0x0 0x0 0x2510113 0x0
juil. 31 02:39:55 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ------------[ DSP dump end ]------------
juil. 31 02:39:55 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: Failed to start DSP
juil. 31 02:39:55 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: failed to boot DSP firmware -110
juil. 31 02:39:55 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: fw_state change: 3 -> 4
juil. 31 02:39:55 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: FW Poll Status: reg[0x4]=0x1d003c timedout
juil. 31 02:39:55 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: hda_dsp_core_reset_enter: timeout on HDA_DSP_REG_ADSPCS read
juil. 31 02:39:55 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: dsp core reset failed: core_mask 1
juil. 31 02:39:55 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: fw_state change: 4 -> 0
juil. 31 02:39:55 tglup-mh kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: sof_probe_work failed err: -110

```

### PR DESCRIPTION
**Pull request by marc-hb**
Fixes #9340

This adds the following commits from the main rimage branch:

```
commit 02abc5d342a3 ("rimage: ace signing functions need openssl 3.0 guards")
commit 73a9d7c897d3 ("rimage: fix openssl 3.0 support in ver25 signing")
commit 8ba3d1729656 ("adsp_config: fix name parsing error in parse_signed_pkg_ace_v1_5")
commit fe4dcaa62c13 ("mtl: Add ASRC module to the manifest")
commit af947cb63dd6 ("config: Add toml config for mtl")
commit c484d99abf89 ("rimage:  add ACE V1.5 handling")
commit 1b233f68bdee ("config: add rmb toml file to support rembrandt build")
```

This is the smallest main branch fast-forward that includes the critical openssl3 fix commit 73a9d7c897d3 ("rimage: fix openssl 3.0 support in ver25 signing") _and_ compiles.

This happens to align the rimage version in stable-v2.2 to the version in stable-v2.3. stable-v2.3 is not in use anymore but it was routinely tested in CI for a long time. In fact this stable-v2.2 commit is the same as stable-v2.3 commit 4e1d3ba61abd3b ("rimage: update to version 02abc5d342a3")

### CODE REVIEW

**Comment by marc-hb** [comment]
https://sof-ci.01.org/sofpr/PR9342/build6806/devicetest/index.html all green.

**Comment by plbossart** [comment]
stable-2.2 should not have any support for ACE and MTL, I am confused here?

**Comment by marc-hb** [comment]
> stable-2.2 should not have any support for ACE and MTL, I am confused here?

Indeed not: here I'm only trying NOT to create a new rimage branch to keep things simple. So this is just a small fast-forward on the old rimage main branch. This happens to pick up a couple irrelevant and harmless commits along the way.

**Comment by plbossart** [comment]
> > stable-2.2 should not have any support for ACE and MTL, I am confused here?
> 
> Indeed not: here I'm trying NOT to create a new rimage branch and avoid cherry-picks to keep things simple. So this is just the smallest possible fast-forward on the old rimage main branch. This happens to pick up a couple irrelevant and harmless commits along the way.

ah ok, i thought those were cherry-picked. Thanks for the explanation.

**Review by cujomalainey — APPROVED** [review]

**Comment by brentlu** [comment]
The test result is promising. FW download success and HDA/DP playback and DMIC record also work.

=> current stable-v2.2
[    7.692364] sof-audio-pci-intel-tgl 0000:00:1f.3: hda_cl_copy_fw: timeout with rom_status_reg (0x80000) read
[    7.692775] sof-audio-pci-intel-tgl 0000:00:1f.3: ------------[ DSP dump start ]------------
[    7.692778] sof-audio-pci-intel-tgl 0000:00:1f.3: Firmware download failed
[    7.692779] sof-audio-pci-intel-tgl 0000:00:1f.3: fw_state: SOF_FW_BOOT_IN_PROGRESS (3)
[    7.692794] sof-audio-pci-intel-tgl 0000:00:1f.3: 0x80000012: module: ROM, state: CSE_VALIDATE_IMAGE_REQUEST, not running
[    7.692811] sof-audio-pci-intel-tgl 0000:00:1f.3: error code: 0x2c (error: signature verification failed)
[    7.692828] sof-audio-pci-intel-tgl 0000:00:1f.3: extended rom status:  0x80000012 0x2c 0x0 0x0 0x0 0x0 0x256053e 0x0
[    7.692830] sof-audio-pci-intel-tgl 0000:00:1f.3: ------------[ DSP dump end ]------------
[    7.692912] sof-audio-pci-intel-tgl 0000:00:1f.3: Failed to start DSP
[    7.692914] sof-audio-pci-intel-tgl 0000:00:1f.3: error: failed to boot DSP firmware -110
[    7.743630] sof-audio-pci-intel-tgl 0000:00:1f.3: error: hda_dsp_core_reset_enter: timeout on HDA_DSP_REG_ADSPCS read
[    7.743636] sof-audio-pci-intel-tgl 0000:00:1f.3: error: dsp core reset failed: core_mask 1
[    8.380548] sof-audio-pci-intel-tgl 0000:00:1f.3: error: sof_probe_work failed err: -110

=> after patched:
[    4.730057] sof-audio-pci-intel-tgl 0000:00:1f.3: Firmware info: version 2:2:10-159b6
[    4.730065] sof-audio-pci-intel-tgl 0000:00:1f.3: Firmware: ABI 3:23:0 Kernel ABI 3:23:0
[    4.730070] sof-audio-pci-intel-tgl 0000:00:1f.3: Firmware debug build 65535 on dtermin.-extman - options:

**Review by plbossart — APPROVED** [review]
